packages/babel-core/src/transformation/normalize-file.js

```diff
@@ -69,12 +69,14 @@ export default function normalizeFile(
     }
   }
 
+  debugger; // eslint-disable-line no-debugger
   if (ast) {
     if (ast.type === "Program") {
       ast = t.file(ast, [], []);
     } else if (ast.type !== "File") {
       throw new Error("AST root must be a Program or File node");
     }
+    ast = t.cloneNode(ast);
```

Comment:
Is this the intended use for t.cloneNode?

Reply:
The problem is that `cloneNode` only clones properties which represents other nodes. It doesn't copy every property (e.g. position, extras...).

I would use lodash's cloneDeep function.

Reply:
Good idea. Will take a look at that.